pkg/osquery/extension.go

```diff
@@ -449,13 +450,20 @@ func (e *Extension) Enroll(ctx context.Context) (string, bool, error) {
 	// If no cached node key, enroll for new node key
 	// note that we set invalid two ways. Via the return, _or_ via isNodeInvaliderr
 	keyString, invalid, err := e.serviceClient.RequestEnrollment(ctx, enrollSecret, identifier, enrollDetails)
-	if isNodeInvalidErr(err) {
+
+	switch {
+	case errors.Is(err, service.ErrDeviceDisabled{}):
+		uninstall.Uninstall(ctx, e.knapsack, true)
```

Comment:
Should we return here? Falling through looks wrong

Reply:
+1 -- I think helpful to add a comment here stating that uninstall will stop launcher immediately so we don't need to do anything further? And maybe do the early return anyway, in case uninstall fails?

Reply:
I'm not sure it needs much comment, but it feels like the node-invalid flow. It's a different action.

(I almost wonder if middleware is a cleaner approach, but I don't want that can of worms)

Reply:
added return